kotsadm/pkg/apiserver/server.go

```diff
@@ -161,3 +168,69 @@ func Start() {
 
 	log.Fatal(srv.ListenAndServe())
 }
+
+// Detects the InstanceID of kodsadm pod across restores
+func generateKotsadmID() error {
+
+	logger.Info("Generate Kotsadm Instance ID")
+
+	// Retrieve the ClusterID from store
+	clusters, err := store.GetStore().ListClusters()
+	if err != nil {
+		logger.Errorf("Failed to list clusters %v", err)
+		return err
+	}
+	if len(clusters) == 0 {
+		logger.Info("No clusters found")
+		return nil
+	}
+	clusterID := clusters[0].ClusterID
+
+	// Write a Query to set/get an Event from the Store
+	keyExists, err := store.GetStore().IsKotsadmIDGenerated()
+	// failed to scan
+	if err != nil {
+		logger.Errorf("Failed to check kotsadm id generation %v", err)
+		return err
+	}
+
+	// if the key does not exist, likely a Restore or pod restart
+	if keyExists == true {
```

Comment:
no need for `== true`
```suggestion
	if keyExists {
```